src/annotator/highlight-clusters.tsx

```diff
@@ -98,6 +109,16 @@ export class HighlightClusterController implements Destroyable {
     this._outerContainer.remove();
   }
 
+  /**
+   * Indicate that we need to update the nesting-related data attributes on all
+   * highlight elements in the document (<hypothesis-highlight> and
+   * SVG rect.hypothesis-svg-highlight elements)
+   */
+  scheduleClusterUpdates() {
+    clearTimeout(this._updateTimeout);
+    this._updateTimeout = setTimeout(() => this._updateClusterData(), 200);
```

Comment:
I think you could shorten this delay for a faster perceived update, as highlighting will typically take less than 200ms.